### PR TITLE
Remove use of Thread.Sleep

### PR DESCRIPTION
**Pull request by westin-m**
Remove unnecessary usage of `Thread.Sleep` in the `UiTestHelpers` class.

### CODE REVIEW

File: tests/E2E Tests/WebAppUiTests/UiTestHelpers.cs [hunk 1]
```diff
@@ -324,7 +324,6 @@ internal static bool StartAndVerifyProcessesAreRunning(List<ProcessStartOptions>
                                                 processDataEntry.EnvironmentVariables);
 
                 processes.Add(processDataEntry.ExecutableName, process);
-                Thread.Sleep(5000);
```
Comment (msbw2) [inline tests/E2E Tests/WebAppUiTests/UiTestHelpers.cs:327]: Is there a way to block the thread on waiting for all processes to start or error? I assume that's the intent behind this sleep.

File: tests/E2E Tests/WebAppUiTests/UiTestHelpers.cs [hunk 2]
```diff
@@ -357,7 +356,6 @@ static void RestartProcesses(Dictionary<string, Process> processes, List<Process
                                                     processDataEntry.AppLocation,
                                                     processDataEntry.ExecutableName,
                                                     processDataEntry.EnvironmentVariables);
-                    Thread.Sleep(5000);
```
Comment (msbw2) [inline tests/E2E Tests/WebAppUiTests/UiTestHelpers.cs:360]: Same here.

**Review by jmprieur — COMMENTED** [review]
Thanks @westin-m 
Did you try this? I think we need to let some time for the web APIs to start?

File: tests/E2E Tests/WebAppUiTests/UiTestHelpers.cs (hunk 1 above)
Comment (bgavrilMS) [reply tests/E2E Tests/WebAppUiTests/UiTestHelpers.cs:327]: If you can check that all processes have started, you can poll on that condition. Like here https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/tests/Microsoft.Identity.Test.Common/TestCommon.cs#L258

Comment (westin-m) [reply tests/E2E Tests/WebAppUiTests/UiTestHelpers.cs:327]: Interesting, it looks like something similar is being done with these tests, after the processes are checked to be running https://github.com/AzureAD/microsoft-identity-web/blob/011bd155ce3ddedeb43daa7348e09584cfa20552/tests/E2E%20Tests/WebAppUiTests/WebAppCallsApiCallsGraphLocally.cs#L335

**Pull request closed without merging**